mlflow/utils/request_utils.py

```diff
@@ -0,0 +1,154 @@
+# DO NO IMPORT MLFLOW IN THIS FILE.
+# This file is imported by download_cloud_file_chunk.py.
+# Importing mlflow is time-consuming and we want to avoid that in artifact download subprocesses.
+import os
+import requests
+import urllib3
+
+from functools import lru_cache
+from packaging.version import Version
+from requests.adapters import HTTPAdapter
+from urllib3.util import Retry
+
+# Response codes that generally indicate transient network failures and merit client retries,
+# based on guidance from cloud service providers
+# (https://docs.microsoft.com/en-us/azure/architecture/best-practices/retry-service-specific#general-rest-and-retry-guidelines)
+_TRANSIENT_FAILURE_RESPONSE_CODES = frozenset(
+    [
+        408,  # Request Timeout
+        429,  # Too Many Requests
+        500,  # Internal Server Error
+        502,  # Bad Gateway
+        503,  # Service Unavailable
+        504,  # Gateway Timeout
+    ]
+)
+
+
+def download_chunk(range_start, range_end, headers, download_path, http_uri):
+    combined_headers = {**headers, "Range": f"bytes={range_start}-{range_end}"}
+
+    with cloud_storage_http_request(
+        "get", http_uri, stream=False, headers=combined_headers
+    ) as response:
+        # File will have been created upstream. Use r+b to ensure chunks
```

Comment:
```suggestion
        augmented_raise_for_status(response)
        # File will have been created upstream. Use r+b to ensure chunks
```

Can we validate the response is successful?

Reply:
We can fix this in a follow-up PR.